### ISSUE TITLE
[Bug]: 执行评估脚本卡住

### ISSUE THREAD

**Issue by jaefan11** [RLinf/RLinf#756]
### Affected component

bash examples/embodiment/eval_embodiment.sh libero_10_ppo_openpi_pi05_eval 执行后卡住 也没有报错

### Bug description

bash examples/embodiment/eval_embodiment.sh libero_10_ppo_openpi_pi05_eval 执行后卡住 也没有报错

### Configuration YAML file

YAML config file:

You can also paste the full config here.
```
defaults:
  - env/libero_10@env.train
  - env/libero_10@env.eval
  - model/pi0_5@actor.model
  - training_backend/fsdp@actor.fsdp_config
  - override hydra/job_logging: stdout

hydra:
  run:
    dir: .
  output_subdir: null
  searchpath:
    - file://${oc.env:EMBODIED_PATH}/config/

cluster:
  num_nodes: 1
  component_placement:
    actor,env,rollout: 4-5
    # env: 0
    # rollout: 1
    # actor: 2

runner:
  task_type: embodied
  logger:
    log_path: "../results"
    project_name: rlinf
    experiment_name: "libero_10_ppo_openpi_pi05_eval"
    logger_backends: ["tensorboard"] # wandb, swanlab

  max_epochs: 1000
  max_steps: -1

  only_eval: True
  val_check_interval: -1
  save_interval: 40

  resume_dir: null # Optional: path to a saved checkpoint directory, such as 'checkpoints/global_step_10'. If not None, it will be used to resume training.
  ckpt_path: null  # Optional: path to a .pt checkpoint. If not None, it will be loaded after the model is instantiated (for evaluation).

algorithm:

  normalize_advantages: True
  kl_penalty: kl  # how to estimate kl divergence: kl or kl_penalty
  group_size: 1
  reward_coef: 1.0
  rollout_epoch: 8
  eval_rollout_epoch: 1 # set eval_rollout_epoch > 0 when enable runner.only_eval or runner.val_check_interval > 0

  reward_type: chunk_level
  logprob_type: chunk_level
  entropy_type: token_level
  update_epoch: 4
  adv_type: gae
  loss_type: actor_critic
  loss_agg_func: "token-mean" 
  kl_beta: 0.0
  entropy_bonus: 0
  clip_ratio_high: 0.2
  clip_ratio_low: 0.2
  clip_ratio_c: 3.0
  value_clip: 0.2
  huber_delta: 10.0

  gamma: 0.99
  gae_lambda: 0.95

  filter_rewards: False
  rewards_lower_bound: 0.1
  rewards_upper_bound: 0.9
  # params for generation
  sampling_params:
    do_sample: False
    temperature_train: 1.0
    temperature_eval: 0.6
    top_k: 50
    top_p: 1.0
    repetition_penalty: 1.0
    add_BOS: False

  # length argument for autoregressive sampling
  # max length means max amount of tokens to generate
  length_params:
    max_new_token: null
    max_length: 1024
    min_length: 1

env:
  group_name: "EnvGroup"
  enable_offload: False

  train:
    total_num_envs: 64
    max_episode_steps: 480 # max episode steps for truncation
    max_steps_per_rollout_epoch: 480
    reward_coef: ${algorithm.reward_coef}
  eval:
    total_num_envs: 50
    auto_reset: True
    ignore_terminations: True
    max_episode_steps: 480
    max_steps_per_rollout_epoch: 480
    group_size: 1
    use_fixed_reset_state_ids: True
    is_eval: True
    video_cfg:
      save_video: True
      video_base_dir: ${runner.logger.log_path}/video/eval

rollout:
  group_name: "RolloutGroup"
  generation_backend: "huggingface"
  recompute_logprobs: False
  unnorm_key: libero_10
  enable_offload: False
  pipeline_stage_num: 1

  model:
    model_path: "/data/code/RLinf/RLinf-Pi05-SFT"
    precision: ${actor.model.precision}

actor:
  group_name: "ActorGroup"
  training_backend: "fsdp"
  micro_batch_size: 128
  global_batch_size: 2048 # maximum : global batch size = micro batch size * actor num_processes.
  seed: 42
  enable_offload: False

  # Override the default values in model/pi0_5
  model:
    model_path: "/data/code/RLinf/RLinf-Pi05-SFT"
    model_type: "openpi"
    add_value_head: True
    # openpi specific parameters
    openpi:
      value_after_vlm: True

  optim:
    lr: 5.0e-6
    value_lr: 1.0e-4
    adam_beta1: 0.9
    adam_beta2: 0.95
    adam_eps: 1.0e-05
    weight_decay: 0.01
    clip_grad: 1.0
    critic_warmup_steps: 0
    total_training_steps: 1000
    min_lr_rate: 0.1
    lr_scheduler: "cosine" # Learning rate scheduler type. Choices: ["constant", "cosine"]. Default: "constant".

  # Override the default values in training_backend/fsdp
  fsdp_config:
    strategy: "fsdp"
    sharding_strategy: "no_shard"
    gradient_checkpointing: False # for openpi, gradient checkpointing is not supported, please do not change this value
    mixed_precision:
      param_dtype: ${actor.model.precision}
      reduce_dtype: ${actor.model.precision}
      buffer_dtype: ${actor.model.precision}

reward:
  use_reward_model: False

critic:
  use_critic_model: False
```

### Log file

/opt/venv/openpi/lib/python3.11/site-packages/hydra/_internal/defaults_list.py:251: UserWarning: In 'libero_10_ppo_openpi_pi05_eval': Defaults list is missing `_self_`. See https://hydra.cc/docs/1.2/upgrades/1.0_to_1.1/default_composition_order for more information
  warnings.warn(msg, UserWarning)
/opt/venv/openpi/lib/python3.11/site-packages/hydra/_internal/hydra.py:119: UserWarning: Future Hydra versions will no longer change working directory at job runtime by default.
See https://hydra.cc/docs/1.2/upgrades/1.1_to_1.2/changes_to_job_working_dir/ for more information.
  ret = run_job(
2026-02-26 08:11:11,582	INFO worker.py:1908 -- Started a local Ray instance. View the dashboard at [1m[32mhttp://127.0.0.1:8265 [39m[22m
[INFO 08:11:16 RLinf] RLinf is running on a cluster with 1 node and 8 accelerators. The nodes' details are: 
node_labels: []
node_rank: 0
ray_id: b17a924e2cbb95c27066160f6d867f8342b37be97b5f48842a40705d
node_ip: 192.168.13.200
num_cpus: 192
python_interpreter_path: /opt/venv/openpi/bin/python
hardware_resources:
- type: Accelerator
  infos:
  - type: Accelerator
    model: NV_GPU:A100-SXM4-80GB
  - type: Accelerator
    model: NV_GPU:A100-SXM4-80GB
  - type: Accelerator
    model: NV_GPU:A100-SXM4-80GB
  - type: Accelerator
    model: NV_GPU:A100-SXM4-80GB
  - type: Accelerator
    model: NV_GPU:A100-SXM4-80GB
  - type: Accelerator
    model: NV_GPU:A100-SXM4-80GB
  - type: Accelerator
    model: NV_GPU:A100-SXM4-80GB
  - type: Accelerator
    model: NV_GPU:A100-SXM4-80GB

Node groups' details are: 
label: cluster
nodes: '0'
hardware_type: Accelerator
ignore_hardware: false
env_configs: null

label: node
nodes: '0'
hardware_type: null
ignore_hardware: true
env_configs: null

[INFO 08:11:16 FlexiblePlacementStrategy] 
[INFO 08:11:16 FlexiblePlacementStrategy] Using flexible placement with hardware ranks: [[4], [5]].
[INFO 08:11:16 FlexiblePlacementStrategy] 
[INFO 08:11:16 FlexiblePlacementStrategy] Using flexible placement with hardware ranks: [[4], [5]].
[INFO 08:11:16 FlexiblePlacementStrategy] 
[INFO 08:11:16 FlexiblePlacementStrategy] Using flexible placement with hardware ranks: [[4], [5]].
[INFO 08:11:16 FlexiblePlacementStrategy] Generated 2 placements: [Placement(rank=0, cluster_node_rank=0, placement_node_rank=0, local_accelerator_rank=4, accelerator_type='NV_GPU', local_rank=0, local_world_size=2, visible_accelerators=['4'], isolate_accelerator=True, local_hardware_ranks=[4], node_group_label='cluster'), Placement(rank=1, cluster_node_rank=0, placement_node_rank=0, local_accelerator_rank=5, accelerator_type='NV_GPU', local_rank=1, local_world_size=2, visible_accelerators=['5'], isolate_accelerator=True, local_hardware_ranks=[5], node_group_label='cluster')].
[INFO 08:11:16 FlexiblePlacementStrategy] Generated 2 placements: [Placement(rank=0, cluster_node_rank=0, placement_node_rank=0, local_accelerator_rank=4, accelerator_type='NV_GPU', local_rank=0, local_world_size=2, visible_accelerators=['4'], isolate_accelerator=True, local_hardware_ranks=[4], node_group_label='cluster'), Placement(rank=1, cluster_node_rank=0, placement_node_rank=0, local_accelerator_rank=5, accelerator_type='NV_GPU', local_rank=1, local_world_size=2, visible_accelerators=['5'], isolate_accelerator=True, local_hardware_ranks=[5], node_group_label='cluster')].
[INFO 08:11:16 NodePlacementStrategy] 
[INFO 08:11:16 NodePlacementStrategy] Using node placement with node ranks: [0].
[INFO 08:11:16 NodePlacementStrategy] Generated 1 placements: [Placement(rank=0, cluster_node_rank=0, placement_node_rank=0, local_accelerator_rank='0', accelerator_type='NV_GPU', local_rank=0, local_world_size=1, visible_accelerators=['0', '1', '2', '3', '4', '5', '6', '7'], isolate_accelerator=True, local_hardware_ranks=[], node_group_label='cluster')].
[INFO 08:11:16 NodePlacementStrategy] 
[INFO 08:11:16 NodePlacementStrategy] Using node placement with node ranks: [0].
[INFO 08:11:16 NodePlacementStrategy] Generated 1 placements: [Placement(rank=0, cluster_node_rank=0, placement_node_rank=0, local_accelerator_rank='0', accelerator_type='NV_GPU', local_rank=0, local_world_size=1, visible_accelerators=['0', '1', '2', '3', '4', '5', '6', '7'], isolate_accelerator=True, local_hardware_ranks=[], node_group_label='cluster')].
2026-02-26 08:11:17.052157: I tensorflow/core/util/port.cc:153] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
2026-02-26 08:11:17.113896: I tensorflow/core/platform/cpu_feature_guard.cc:210] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 AVX512F AVX512_VNNI AVX512_BF16 AVX512_FP16 AVX_VNNI AMX_TILE AMX_INT8 AMX_BF16 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
/opt/venv/openpi/lib/python3.11/site-packages/keras/src/export/tf2onnx_lib.py:8: FutureWarning: In the future `np.object` will be defined as the corresponding NumPy scalar.
  if not hasattr(np, "object"):
[WARNING 08:11:20 RLinf] ---------init_workers-----1-
[36m(EnvWorker pid=846252)[0m [INFO 08:11:20 FlexiblePlacementStrategy] 
[36m(EnvWorker pid=846252)[0m [INFO 08:11:20 FlexiblePlacementStrategy] Using flexible placement with hardware ranks: [[4], [5]].
[36m(EnvWorker pid=846254)[0m [INFO 08:11:20 FlexiblePlacementStrategy] 
[36m(EnvWorker pid=846254)[0m [INFO 08:11:20 FlexiblePlacementStrategy] Using flexible placement with hardware ranks: [[4], [5]].
[36m(MultiStepRolloutWorker pid=846241)[0m [INFO 08:11:21 FlexiblePlacementStrategy] 
[36m(MultiStepRolloutWorker pid=846241)[0m [INFO 08:11:21 FlexiblePlacementStrategy] Using flexible placement with hardware ranks: [[4], [5]].
[36m(MultiStepRolloutWorker pid=846250)[0m [INFO 08:11:21 FlexiblePlacementStrategy] 
[36m(MultiStepRolloutWorker pid=846250)[0m [INFO 08:11:21 FlexiblePlacementStrategy] Using flexible placement with hardware ranks: [[4], [5]].
[36m(MultiStepRolloutWorker pid=846250)[0m /opt/venv/openpi/lib/python3.11/site-packages/keras/src/export/tf2onnx_lib.py:8: FutureWarning: In the future `np.object` will be defined as the corresponding NumPy scalar.
[36m(MultiStepRolloutWorker pid=846250)[0m   if not hasattr(np, "object"):
[36m(MultiStepRolloutWorker pid=846241)[0m /opt/venv/openpi/lib/python3.11/site-packages/keras/src/export/tf2onnx_lib.py:8: FutureWarning: In the future `np.object` will be defined as the corresponding NumPy scalar.
[36m(MultiStepRolloutWorker pid=846241)[0m   if not hasattr(np, "object"):
[36m(MultiStepRolloutWorker pid=846241)[0m INFO:datasets:PyTorch version 2.6.0 available.
[36m(MultiStepRolloutWorker pid=846241)[0m INFO:datasets:Polars version 1.36.1 available.
[36m(MultiStepRolloutWorker pid=846241)[0m INFO:datasets:TensorFlow version 2.20.0 available.
[36m(MultiStepRolloutWorker pid=846241)[0m INFO:datasets:JAX version 0.5.3 available.
[36m(MultiStepRolloutWorker pid=846250)[0m INFO:datasets:PyTorch version 2.6.0 available.
[36m(MultiStepRolloutWorker pid=846250)[0m INFO:datasets:Polars version 1.36.1 available.
[36m(MultiStepRolloutWorker pid=846250)[0m INFO:datasets:TensorFlow version 2.20.0 available.
[36m(MultiStepRolloutWorker pid=846250)[0m INFO:datasets:JAX version 0.5.3 available.
[36m(MultiStepRolloutWorker pid=846241)[0m INFO:root:Loaded norm stats from /data/code/RLinf/RLinf-Pi05-SFT/physical-intelligence/libero
[36m(MultiStepRolloutWorker pid=846241)[0m INFO:root:Loaded norm stats from /data/code/RLinf/RLinf-Pi05-SFT/physical-intelligence/libero
[36m(MultiStepRolloutWorker pid=846250)[0m INFO:root:Loaded norm stats from /data/code/RLinf/RLinf-Pi05-SFT/physical-intelligence/libero
[36m(MultiStepRolloutWorker pid=846250)[0m INFO:root:Loaded norm stats from /data/code/RLinf/RLinf-Pi05-SFT/physical-intelligence/libero
[WARNING 08:12:28 RLinf] ---------init_workers-----2-
[36m(EnvWorker pid=846252)[0m Gym has been unmaintained since 2022 and does not support NumPy 2.0 amongst other critical functionality.
[36m(EnvWorker pid=846252)[0m Please upgrade to Gymnasium, the maintained drop-in replacement of Gym, or contact the authors of your software and request that they upgrade.
[36m(EnvWorker pid=846252)[0m Users of this version of Gym should be able to simply replace 'import gym' with 'import gymnasium as gym' in the vast majority of cases.
[36m(EnvWorker pid=846252)[0m See the migration guide at https://gymnasium.farama.org/introduction/migration_guide/ for additional information.
[36m(EnvWorker pid=846252)[0m [robosuite WARNING] No private macro file found! (macros.py:53)
[36m(EnvWorker pid=846252)[0m WARNING:robosuite_logs:No private macro file found!
[36m(EnvWorker pid=846252)[0m [robosuite WARNING] It is recommended to use a private macro file (macros.py:54)
[36m(EnvWorker pid=846252)[0m WARNING:robosuite_logs:It is recommended to use a private macro file
[36m(EnvWorker pid=846252)[0m [robosuite WARNING] To setup, run: python /opt/venv/openpi/lib/python3.11/site-packages/robosuite/scripts/setup_macros.py (macros.py:55)
[36m(EnvWorker pid=846252)[0m WARNING:robosuite_logs:To setup, run: python /opt/venv/openpi/lib/python3.11/site-packages/robosuite/scripts/setup_macros.py
[36m(EnvWorker pid=846254)[0m Gym has been unmaintained since 2022 and does not support NumPy 2.0 amongst other critical functionality.
[36m(EnvWorker pid=846254)[0m Please upgrade to Gymnasium, the maintained drop-in replacement of Gym, or contact the authors of your software and request that they upgrade.
[36m(EnvWorker pid=846254)[0m Users of this version of Gym should be able to simply replace 'import gym' with 'import gymnasium as gym' in the vast majority of cases.
[36m(EnvWorker pid=846254)[0m See the migration guide at https://gymnasium.farama.org/introduction/migration_guide/ for additional information.
[36m(EnvWorker pid=846254)[0m [robosuite WARNING] No private macro file found! (macros.py:53)
[36m(EnvWorker pid=846254)[0m WARNING:robosuite_logs:No private macro file found!
[36m(EnvWorker pid=846254)[0m [robosuite WARNING] It is recommended to use a private macro file (macros.py:54)
[36m(EnvWorker pid=846254)[0m WARNING:robosuite_logs:It is recommended to use a private macro file
[36m(EnvWorker pid=846254)[0m [robosuite WARNING] To setup, run: python /opt/venv/openpi/lib/python3.11/site-packages/robosuite/scripts/setup_macros.py (macros.py:55)
[36m(EnvWorker pid=846254)[0m WARNING:robosuite_logs:To setup, run: python /opt/venv/openpi/lib/python3.11/site-packages/robosuite/scripts/setup_macros.py
[36m(EnvWorker pid=846252)[0m 2026-02-26 08:12:29.121702: I tensorflow/core/util/port.cc:153] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
[36m(EnvWorker pid=846254)[0m 2026-02-26 08:12:29.098723: I tensorflow/core/util/port.cc:153] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.

### Environment

python -V:
uv pip list:
nvidia-smi:
RLinf version:
Docker image tag: 

### Before submitting a new issue...

- [x] Have you checked relevant issues, FAQs (https://rlinf.readthedocs.io/en/latest/rst_source/faq.html), or asked the chatbot at the top right corner of the [documentation page](https://rlinf.readthedocs.io).
